### ISSUE TITLE
Improve `name` scope in typescript for multi-initialization statements

### ISSUE THREAD

**Issue by pokey** [pokey/cursorless-vscode#1742]
# The problem

In code like the following:

```ts
const aaa = 0, bbb = 1;
```

we could improve our support for the `name` scope.

# The solution

We should probably handle the case of a simple initializer with only one variable separately from a multi-child one

## Single-variable initializer

For example:

```ts
const aaa = 0;
```

In this case, we want the following:

- A single scope whose domain is the entire statement

This way you could say:

- `"name"` from within the statement to get the name
- `"every name"` from anywhere in the block containing the statement (including within statement itself) to get all the names in that block

## Multi-variable initializer

For example:

```ts
const aaa = 0, bbb = 1;
```

In this case, we want the following:

- one scope per nested assignment (`aaa = 0` and `bbb = 1`)
- a parent scope whose domain is the entire statement, and has multiple content ranges, one per sub-assignment name
- an iteration scope that is the entire statement

This way you could say:

- `"name"` from within a sub-assignment to get the name of that sub-assignment
- `"name"` from the start / end of the statement to get all the names in that statement
- `"every name"` from anywhere in the statement to get all the names in that statement
- `"first name"` from anywhere in the statement to get the first name in that statement 
- `"every name"` from anywhere in the block containing the statement, except within statement itself, to get all the names in that block

# Discussion

Note that we may want to use this pattern in other places

- Talon condition blocks
- Go return values
- Tags in `.scm` files

**Comment by auscompgeek** [comment]
Would you also want the same sort of thing in Python tuple unpacking?

**Comment by pokey** [comment]
Good question. My instinct is to say no. For that it's generally more useful to just target the entire left-hand side. But I think @josharian wants something like that for Go, so maybe we should discuss

**Comment by josharian** [comment]
Yeah, in Go, code like this is very common:

```go
x, err := foo()
```

and wanting to refer to just `x` or `err` individually is also going to be very common.

(It's not tuple unpacking, it's multiple assignment, but it's very similar.)

**Comment by pokey** [comment]
I guess you'd want to be able to say eg "chuck first name", "pour last name", etc?

**Comment by josharian** [comment]
Yeah, after a refactor that changes what a function returns, it's going to be very common to want to chuck or pour or clone or swap individual names.
